test/unit/org/apache/cassandra/utils/CassandraGenerators.java

```diff
@@ -413,10 +491,250 @@ public Gen<KeyspaceMetadata> build()
+            if (nextBoolean(rnd))
+                options.put(Controller.FLUSH_SIZE_OVERRIDE_OPTION, storageSizeGen.generate(rnd));
+            if (nextBoolean(rnd))
+                options.put(Controller.MAX_SSTABLES_TO_COMPACT_OPTION, SourceDSL.integers().between(0, 32).generate(rnd).toString());
+            if (nextBoolean(rnd))
+                options.put(Controller.SSTABLE_GROWTH_OPTION, SourceDSL.integers().between(0, 100).generate(rnd) + "%");
+            if (nextBoolean(rnd))
+                options.put(Controller.OVERLAP_INCLUSION_METHOD_OPTION, SourceDSL.arbitrary().enumValues(Overlaps.InclusionMethod.class).generate(rnd).name());
+            if (nextBoolean(rnd))
+            {
+                int numLevels = SourceDSL.integers().between(1, 10).generate(rnd);
+                String[] scalingParams = new String[numLevels];
+                Gen<Integer> levelSize = SourceDSL.integers().between(2, 10);
+                for (int i = 0; i < numLevels; i++)
+                {
+                    String value;
+                    switch (SourceDSL.integers().between(0, 3).generate(rnd))
+                    {
+                        case 0:
+                            value = "N";
+                            break;
+                        case 1:
+                            value = "L" + levelSize.generate(rnd);
+                            break;
+                        case 2:
+                            value = "T" + levelSize.generate(rnd);
+                            break;
+                        case 3:
+                            value = SourceDSL.integers().all().generate(rnd).toString();
+                            break;
+                        default:
+                            throw new AssertionError();
+                    }
+                    scalingParams[i] = value;
+                }
+                options.put(Controller.SCALING_PARAMETERS_OPTION, String.join(",", scalingParams));
+            }
+            if (nextBoolean(rnd))
+            {
+                // Calculate TARGET then compute the MIN from that.  The issue is that there is a hidden relationship
+                // between these 2 fields more complex than simple comparability, MIN must be < 70% * TARGET!
+                // See CASSANDRA-20398
+                // 1MiB to 128MiB target
+                long targetBytes = SourceDSL.longs().between(1L << 20, 1L << 27).generate(rnd);
+                long limit = (long) Math.ceil(targetBytes * Math.sqrt(0.5));
+                long minBytes = SourceDSL.longs().between(1, limit - 1).generate(rnd);
+                options.put(Controller.MIN_SSTABLE_SIZE_OPTION, minBytes + "B");
+                options.put(Controller.TARGET_SSTABLE_SIZE_OPTION, targetBytes + "B");
+            }
+            return options;
+        };
+        public Gen<CompactionParams> build()
```

Comment:
There are a few other options in AbstractCompactionStrategy that are missing here, like tombstone_threshold

Reply:
added `tombstone_threshold`, `unchecked_tombstone_compaction`, and `only_purge_repaired_tombstones`